reconcile/terraform_tgw_attachments.py

```diff
@@ -56,12 +58,46 @@ class ValidationError(Exception):
     pass
 
 
+class AccountWithAssumeRole(ClusterPeeringConnectionAccountTGWV1_AWSAccountV1):
```

Comment:
Role assumption information is not something, that makes a _ClusterPeeringConnectionAccountTGWV1_AWSAccountV1_ more specific. Would prefer an association between _AccountWithAssumeRole_ and _ClusterPeeringConnectionAccountTGWV1_AWSAccountV1_
Makes the creation of AccountWithAssumeRole easier and does prevent misusage of this dataclass.

Reply:
@janboll I checked all usages of current account object:
* [get_tgws_details](https://github.com/app-sre/qontract-reconcile/blob/e0fae173f22e7a87818c6987e3a00e41cf049934/reconcile/utils/aws_api.py#L1102): `name uid`
* [get_cluster_vpc_details](https://github.com/app-sre/qontract-reconcile/blob/e0fae173f22e7a87818c6987e3a00e41cf049934/reconcile/utils/aws_api.py#L895): `name assume_role assume_region assume_cidr`
* [populate_additional_providers](https://github.com/app-sre/qontract-reconcile/blob/973246d2378a55b8577027731ce4b00893608ce3/reconcile/utils/terrascript_aws_client.py#L841): `name assume_role assume_region`

So instead of treat `AccountWithAssumeRole` as a subclass of `ClusterPeeringConnectionAccountTGWV1_AWSAccountV1`, now it's just a standalone class

```python
class AccountWithAssumeRole(BaseModel):
    name: str
    uid: str
    assume_role: str
    assume_region: str
    assume_cidr: str
```

Reply:
Thank you for looking into this.